hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/pipeline/PipelineManagerImpl.java

```diff
@@ -419,6 +421,46 @@ public void closePipeline(Pipeline pipeline, boolean onTimeout)
     }
   }
 
+  /** close the pipelines whose nodes' IPs are stale.
+   *
+   * @param datanodeDetails new datanodeDetails
+   */
+  @Override
+  public void closeStalePipelines(DatanodeDetails datanodeDetails) {
```

Comment:
I don't see a unit test for this.  Is it not worth it?

Reply:
Added a unit test

Reply:
thanks!